### ISSUE TITLE
Confusing error message when Lmod can't find module

### ISSUE THREAD

**Issue by tobbez** [hpcugent/easybuild-framework#2059]
Stumbled upon this when writing a `ModuleNamingScheme`.

If Lmod can't find the module during the sanity check step, EasyBuild will fail with the message:
```
ERROR EasyBuild crashed with an error (at ?:124 in __init__): Sanity check failed: loading fake module failed: 'Changing environment as dictated by module failed: name \'false\' is not defined
```
The complaint about `false` being undefined is a bit of a red herring -- most often these exception mean that there's a problem with the software raising it. This means I had to spend some time digging through the code to make sure this was not the case.

If you check the full log file you see the output from `lmod`. Its stderr tells us the real problem (but doesn't really dispel the confusion introduced by EasyBuild's error message):
```
Lmod has detected the following error:  The following module(s) are unknown: "[redacted]"

Please check the spelling or version number. Also try "module spider ..."
```
And stdout provides the cause of EasyBuild's error message:
```
os.environ["LMOD_DEFAULT_MODULEPATH"] = "[redacted]";
os.environ["MODULEPATH"] = "[redacted]";
false
```
The final line, `false`, which causes an error when the above is executed as code by `exec` in `run_module` from `easybuild/tools/modules.py`.

It should be possible to do better, by using either Lmod's stderr output, or (likely) more preferably, its exit code (both of which are currently disregarded by EasyBuild).

**Comment by boegel** [comment]
@tobbez the `false` is a trick done by Lmod to make the `module` function (which `eval`s the stdout produced by Lmod) fail

But you're very right, we can do a lot better here. I'm actually wondering why we're not checking the `lmod` exit code at all, that's just wrong, we should be checking it...

Thanks for bringing this up!

**Comment by boegel** [comment]
Checking of exit code on module commands in implemented in #2200

**Comment by boegel** [comment]
Closing this since #2200 is merged.
